src/components/PiecesPreviewModal/PiecesPreviewModal.js

```diff
@@ -226,25 +243,53 @@ const PiecesPreviewModal = ({
   };
 
   return (
-    <FormModal
-      modalProps={{
-        onClose: closeModal,
-        open: showModal,
-        label: (
-          <FormattedMessage id="ui-serials-management.ruleset.generatePredictedPieces" />
-        ),
-        footer: renderFooter,
-      }}
-      mutators={arrayMutators}
-      onSubmit={handleCreation}
-    >
-      <PiecesPreviewModalForm
-        allowCreation={allowCreation}
-        pieceSets={pieceSets}
-        ruleset={ruleset}
+    <>
+      <FormModal
+        modalProps={{
+          onClose: closeModal,
+          open: showModal,
+          label: (
+            <FormattedMessage id="ui-serials-management.ruleset.generatePredictedPieces" />
+          ),
+          footer: renderFooter,
+        }}
+        mutators={arrayMutators}
+        onSubmit={handleCreation}
+      >
+        <PiecesPreviewModalForm
+          allowCreation={allowCreation}
+          pieceSets={pieceSets}
+          ruleset={ruleset}
+        />
+        {dateExists && (
+          <MessageBanner type="warning">
+            <FormattedMessage
+              id="ui-serials-management.pieceSets.overlappingDates.warning"
+              values={{ startDate: intl.formatDate(fieldValues.startDate), serialName }}
+            />
+          </MessageBanner>
```

Comment:
Here we could also place the message banner within the `PiecesPreviewModalForm` component (or within its own component and have it here so that we have access to the `useFormState` hook as opposed to `fieldValues`